test/txsim/account.go

```diff
@@ -225,11 +227,29 @@ func (am *AccountManager) Submit(ctx context.Context, op Operation) error {
 
 	opts := make([]user.TxOption, 0)
 	if op.GasLimit == 0 {
-		opts = append(opts, user.SetGasLimit(DefaultGasLimit), user.SetFee(defaultFee))
+		if am.gasLimit > 0 {
+			// Use the custom gas limit set on the AccountManager
+			opts = append(opts, user.SetGasLimit(am.gasLimit))
+			
+			// Calculate fee based on gas price
+			var gasPrice float64
+			if am.gasPrice > 0 {
+				gasPrice = am.gasPrice
```

Comment:
[nit] 
maybe add a log specifying the custom gas price chosen, especially if it's `< appconsts.DefaultMinGasPrice`. To avoid having to assist people setting values lower than `appconsts.DefaultMinGasPrice` and complaining why it doesn't work

Reply:
thank you! https://github.com/celestiaorg/celestia-app/pull/4447/commits/36d2c38486f3494a6426ea38fb5a481f7a454153